core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1754,9 +1754,14 @@ public void removeNode(@Nonnull Node n) throws IOException {
     /**
      * Updates an existing {@link Node} on disk.
      * @see Nodes#updateNode
+     * @since 1.634
      */
     public boolean updateNode(Node n) throws IOException {
-        return nodes.updateNode(n);
+        if ( nodes.updateNode(n) ){
```

Comment:
@jtnord I had a doubt about it because it will create a cycle Jenkins uses Nodes and Nodes is using Jenkins.instance to access to the queue ... WDYT ?